File: crates/breez-sdk/bindings/examples/cli/langs/react-native/src/contacts.ts
/**
 * Contacts subcommands.
 *
 * Note: Contact management (add, update, delete, list) is not yet supported
 * in the React Native SDK. These handlers return a "not yet supported" message.
 */

/** All contacts subcommand names for help and completion. */
export const CONTACTS_COMMAND_NAMES = [
  'add',
  'update',
  'delete',
  'list',
]

/**
 * Dispatch a contacts subcommand.
 *
 * @param args - The arguments after "contacts" (e.g., ["add", "Alice", "alice@example.com"])
 * @param _sdk - The BreezSdk instance (unused - contacts not yet supported)
 * @returns A string result to display
 */
export async function dispatchContactsCommand(
  args: string[],
  _sdk: unknown
): Promise<string> {
  if (args.length === 0 || args[0] === 'help') {
    return printContactsHelp()
  }

  const subcommand = args[0]

  switch (subcommand) {
    case 'add':
    case 'update':
    case 'delete':
    case 'list':
      return 'Not yet supported in React Native'
    default:
      return `Unknown contacts subcommand: ${subcommand}. Use 'contacts help' for available commands.`
  }
}

function printContactsHelp(): string {
  const lines = [
    '',
    'Contacts subcommands (not yet supported in React Native):',
    '  contacts add <name> <payment_identifier>           Add a new contact',
    '  contacts update <id> <name> <payment_identifier>   Update an existing contact',
    '  contacts delete <id>                               Delete a contact',
    '  contacts list [<offset> <limit>]                   List contacts',
    '',
  ]
  return lines.join('\n')
}

File: crates/breez-sdk/bindings/examples/cli/langs/flutter/lib/contacts.dart
import 'package:breez_sdk_spark_flutter/breez_sdk_spark.dart';

/// Contacts subcommand names (used for tab completion).
const contactsCommandNames = ['contacts add', 'contacts update', 'contacts delete', 'contacts list'];

typedef ContactsHandler = Future<void> Function(BreezSdk sdk, List<String> args);

class _ContactsEntry {
  final String description;
  final ContactsHandler handler;
  const _ContactsEntry(this.description, this.handler);
}

Map<String, _ContactsEntry>? _registry;

Map<String, _ContactsEntry> _getRegistry() {
  return _registry ??= {
    'add': _ContactsEntry('Add a new contact', _handleAdd),
    'update': _ContactsEntry('Update an existing contact', _handleUpdate),
    'delete': _ContactsEntry('Delete a contact', _handleDelete),
    'list': _ContactsEntry('List contacts', _handleList),
  };
}

/// Dispatch a contacts subcommand given the args after 'contacts'.
Future<void> dispatchContactsCommand(List<String> args, BreezSdk sdk) async {
  final registry = _getRegistry();

  if (args.isEmpty || args[0] == 'help' || args[0] == '--help') {
    print('\nContacts subcommands (not yet supported in Flutter):\n');
    for (final entry in registry.entries.toList()..sort((a, b) => a.key.compareTo(b.key))) {
      print('  contacts ${entry.key.padRight(30)} ${entry.value.description}');
    }
    print('');
    return;
  }

  final subName = args[0];
  final subArgs = args.sublist(1);

  if (!registry.containsKey(subName)) {
    print("Unknown contacts subcommand: $subName. Use 'contacts help' for available commands.");
    return;
  }

  await registry[subName]!.handler(sdk, subArgs);
}

// --- add ---

Future<void> _handleAdd(BreezSdk sdk, List<String> args) async {
  print('Not yet supported');
}

// --- update ---

Future<void> _handleUpdate(BreezSdk sdk, List<String> args) async {
  print('Not yet supported');
}

// --- delete ---

Future<void> _handleDelete(BreezSdk sdk, List<String> args) async {
  print('Not yet supported');
}

// --- list ---

Future<void> _handleList(BreezSdk sdk, List<String> args) async {
  print('Not yet supported');
}
